src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheckTest.java

```diff
@@ -0,0 +1,233 @@
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.Assert.fail;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.DetailAstImpl;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+public class MatchXpathCheckTest
+    extends AbstractModuleTestSupport {
+
+    @Override
+    protected String getPackageLocation() {
+        return "com/puppycrawl/tools/checkstyle/checks/coding/matchxpath";
+    }
+
+    @Test
+    public void testCheckWithEmptyQuery()
+            throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(MatchXpathCheck.class);
+        final String[] expected = {};
+        verify(checkConfig, getPath("InputMatchXpath.java"), expected);
+    }
+
+    @Test
+    public void testCheckWithImplicitEmptyQuery()
+            throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(MatchXpathCheck.class);
+        checkConfig.addAttribute("query", "");
+        final String[] expected = {};
+        verify(checkConfig, getPath("InputMatchXpath.java"), expected);
+    }
+
+    @Test
+    public void testCheckWithMatchingMethodNames()
+            throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(MatchXpathCheck.class);
+        checkConfig.addAttribute("query", "//METHOD_DEF[./IDENT[@text='test' or @text='foo']]");
+        final String[] expected = {
+            "8:5: " + getCheckMessage(MatchXpathCheck.MSG_KEY),
+            "10:5: " + getCheckMessage(MatchXpathCheck.MSG_KEY),
+        };
+        verify(checkConfig, getPath("InputMatchXpath.java"), expected);
+    }
+
+    @Test
+    public void testCheckWithNoMatchingMethodName()
+            throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(MatchXpathCheck.class);
+        checkConfig.addAttribute("query", "//METHOD_DEF[./IDENT[@text='wrongName' or "
+                + "@text='nonExistingMethod']]");
+        final String[] expected = {};
```

Comment:
Just a suggestion for consistency, most Checkstyle UT's use `final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;`

Reply:
Same for line 45, 55

Reply:
Thank you very much :) 

Fixed mentioned lines